### PR TITLE
Add Symfony's WebProfiler

### PR DESCRIPTION
**Pull request by Pierstoval**

### CODE REVIEW

**Comment by cconard96** [comment]
How much does this duplicate the work already done by the GLPI debug bar/profiler/debug profiles (#14594)?

**Comment by Pierstoval** [comment]
The goal is to be able to debug the newly integrated Symfony kernel. The current debugging process is GLPI-specific. Integrating Symfony's profiler is a first step towards having a full-fledged debugger including both GLPI's one and the Symfony one :)

(also, I'm using it to debug the ongoing work on #17361 )

You might note the fact that it's not pointing on `main` but on the `feature/advanced-routing` branch

File: dependency_injection/services.php [hunk 1]
```diff
@@ -40,6 +40,12 @@
 return static function (ContainerConfigurator $container): void {
     $projectDir = dirname(__DIR__);
 
+    $parameters = $container->parameters();
+
+    // Default secret, just in case
+    $parameters->set('env(APP_SECRET_FILE)', $projectDir . '/config/glpicrypt.key');
+    $parameters->set('kernel.secret', env('file:APP_SECRET_FILE'));
```
Comment (cedric-anne) [inline dependency_injection/services.php:47]: This file will not exist on a fresh installation, until its creation at the end of the installation process. What will be put in `kernel.secret` when the file does not exist ? Will it produce an error ?

Comment (Pierstoval) [reply dependency_injection/services.php:47]: The parameter is used only in the Web Profiler (not even the debug toolbar), and Symfony will only fetch the variable at runtime (so if you open the Profiler). And you can't open the Profiler if the project isn't installed anyway...

Also, the installation process won't use the profiler, so it's safe for now.

**Comment by cedric-anne** [comment]
To be clear, the Symfony debug/profiler tools will never be packaged in the GLPI release archive. They must never be available in a production environment as they may expose sensitive information.

The GLPI debug bar will still be available in the production environment, when debug mode is active, at least in the short term. I would like to remove the GLPI debug bar from the production environment, but I am afraid many people will complain about such removal.

**Comment by cconard96** [comment]
> I am afraid many people will complain about such removal.

Me included. Probably not a big deal for paid support/cloud cases, but I would not be happy or very helpful helping community members debug something when they cannot provide the details given to them in the debug bar and that isn't fair to community members.

File: dependency_injection/services.php (hunk 1 above)
Comment (cedric-anne) [reply dependency_injection/services.php:47]: Then maybe defaulting to null is a better solution, if we want to be able, at some point, to use the profiler to debug the installation process.
```suggestion
    $parameters->set('kernel.secret', env('default::file:APP_SECRET_FILE'));
```

**Comment by Pierstoval** [comment]
That said, the only sensitive information the Symfony Profiler displays is the `phpinfo`, accessible at `/_profiler/phpinfo`, which we can disable by overwriting the route.

It's even possible to disable some Symfony collectors depending on the environment.

However, allowing it to be disabled/enabled while the app is running (so with a database option) is not recommended. It could be done with an env var, a specific file, or something external, but it really has to be tested beforehand.

This PR only enables the debugger in the `development` kernel environment anyway, so it's fine so far

File: dependency_injection/services.php (hunk 1 above)
Comment (Pierstoval) [reply dependency_injection/services.php]: Can't use `null`, it has to be a string. I created a default value that's stored during the container compilation process (then it happens only once)

**Review by cedric-anne — APPROVED** [review]

**Comment by Pierstoval** [comment]
🚀

**Pull request merged**